docs/adr/adr-00XX-making-the-component-theme-customisable.md

```diff
@@ -0,0 +1,246 @@
+
+#### Updating the build process
+
+Before that, we did not use CSS files.
+We had to update the build process so that the CSS file would now be part of the final bundle both for the CDN and the npm package.
+
+##### NPM Package
+
+For the npm package, we decided to copy the CSS file inside the "styles" folder.
+
+To do so, we rely on the [rollup plugin copy](https://www.npmjs.com/package/rollup-plugin-copy).
+We also minify the CSS file even though it is really small for the moment.
+
+Rollup npm config:
+```js
+  plugins: [
+    ...,
+    copy({
+      targets: [
+        {
+          src: 'src/styles/default-theme.css',
+          dest: 'dist/styles',
+          transform: (stylesheet) => minifyStylesheet(stylesheet),
+        },
+      ],
+    }),
+  ]
+```
+
+To use the components inside a project, one needs to import this CSS file from `node_modules` as explained in the [how to use the components theme](../guides/how-to-use-the-components-theme.md).
+
+##### CDN
+
+###### Caching system
+
+Updating the CDN so that it would serve the CSS file was slightly more complicated.
+Since the default theme might not change very often, we thought it was useful to take advantage of the caching system.
+Let's imagine the following case:
+
+* we publish version 9.0.0 of our components, introducing the `default-theme` CSS file.
+* a few weeks later we publish version 9.1.0 of our components. We did not change anything in the `default-theme` CSS file.
+
+We want the CDN to serve the same CSS file for both the version 9.0.0 and 9.1.0 so that people that were using the 9.0.0 version don't have to fetch a new file for nothing.
+
+The CDN already uses a system to do this exact thing for components.
+Filenames contain a hash that is generated by rollup based on their content.
+This means their filenames only change if their content changes.
+
+###### Rollup plugins and deps manifest
+
+To be given a filename with a hash, the file needs to be processed by rollup as part of the bundle so we could not rely on the copy rollup plugin like we did for the npm build.
+
+Rollup typically only processes JavaScript files. 
+This means our new CSS file was not part of its bundle so it was neither copied nor given a filename with a hash based on its content.
+
+That is why we created a custom plugin named `stylesAssetsPlugin`.
+This plugin reads the CSS file, minifies it, and uses the rollup `emitFile` function to generate a CSS file that is part of the rollup bundle.
+
+```js
+const stylesheet = fs.readFileSync('src/styles/default-theme.css', 'utf8');
+const minifiedStylesheet = transform(stylesheet);
+
+this.emitFile({
+  type: 'asset',
+  name: 'default-theme.css',
+  source: minifiedStylesheet,
+});
+```
+
+Rollup generates a filename with a hash based on the file content.
+
+Then, the file needs to be referenced inside the manifest to be associated with a given components version (9.0.0 for instance).
+This is done inside the `deps-manifest` plugin.
```

Comment:
Maybe we should describe what is the `deps-manifest`. Is there a doc somewhere that we could reference here? If not, maybe a simple description of what it is could help the reader

Reply:
Yes you are definitely right, I'll try to come up with a good synthetic explanation ! Thanks :heart:

Reply:
Ok so I tried to explain it without going into details about how it works.

@pdesoyres-cc What do you think about this explanation ? Do you find it clear enough or do you need more details ?

I tried to explain more at first but it ends up being way too long and very specific, to the point that I think we might need a proper ADR about the CDN, its manifest, etc. (and I think it's a very interesting subject).

Reply:
I discussed this morning with our awesome @hsablonniere colleague about writing an ADR on that subject. Maybe when its done we will add a reference to this ADR.

I like your proposal but maybe we could simplify it with:

```
Then, this hashed file name needs to be associated with the right version (9.0.0 for instance).
This is done inside our homemade `deps-manifest` rollup plugin.
```

Reply:
Looks great, I'll use your suggestion. Thanks !

Reply:
I really need to write about this. (I gave two talks about it but a written ADR is needed :p)